### DISCUSSION TITLE
XYPlot with only one x/y value pair

### DISCUSSION THREAD

**Discussion by axkr** [jfree/jfreechart#314] [Q&A] [answered]
I'm trying to fix a problem if a `ListLogPlot({2147483647})` has only one value pair (i.e. [1, 2147483647])
I get the following `IllegalArgumentException`.

How can I solve this by manually configuring "the range"?

```java
java.lang.IllegalArgumentException: A positive range length is required: Range[2.1474836469999974E9,2.1474836469999974E9]
	at org.jfree.chart.axis.ValueAxis.setRange(ValueAxis.java:1198)
	at org.jfree.chart.axis.LogAxis.autoAdjustRange(LogAxis.java:470)
	at org.jfree.chart.axis.LogAxis.configure(LogAxis.java:420)
	at org.jfree.chart.plot.XYPlot.setRangeAxis(XYPlot.java:877)
	at org.matheclipse.core.bridge.fig.ListPlot.listPlot(ListPlot.java:89)
```

jfreechart 1.5.3 is used.

**Comment by trashgod** [comment]
I'm not sure what `ListLogPlot` is, but how does the range come to have identical lower and upper bounds? The `Range` in the stack trace has a length of zero, which results in the exception.

    System.out.println(new Range(2.1474836469999974E9,2.1474836469999974E9).getLength());

Console:

    0.0

**Reply by axkr** [reply]
I want to draw a chart with only one value pair: `x=1.0` and `y=2.1474836469999974E9` and the y-axis should be logarithmic.

Because this is data directly from the users input, I cannot control the number of value pairs coming in (or I have to display an error, but why?)
Do you have a simple example how I can do it?

**Comment by trashgod** [comment, marked as answer]
I used the `LogAxis` example examined [here](https://stackoverflow.com/q/70618914/230513) and enabled renderer shapes as suggested [here](https://stackoverflow.com/a/73312999/230513) to get the chart pictured below:

    series.add(1, Math.E);
    …
    //yAxis.setStandardTickUnits(NumberAxis.createIntegerTickUnits());
    XYPlot plot = new XYPlot(new XYSeriesCollection(series),
        xAxis, yAxis, new XYLineAndShapeRenderer(true, true));

<img width="752" alt="Log" src="https://user-images.githubusercontent.com/32623871/189176662-25408c06-fd3d-44e0-b361-eab120812fa8.png">

**Reply by axkr** [reply]
Thank you I will try this approach.
